client/src/tutorials/ZeroCopy/pages/1.md

```diff
@@ -0,0 +1,179 @@
+# How to handle big accounts
+
+This example will show you how to create big accounts on Solana and give you an easy playing field to try out different things.
+Because big accounts are expensive you will need to run the tests against a local validator:
+
+## How to run
+
+To run the tests:
+
+1. Install Solana CLI: https://docs.solana.com/de/cli/install-solana-cli-tools
+2. Open terminal and type solana-test-validator to start a local validator
+3. In the settings on the bottom left pick provider local.
+4. Build and deploy the program
+5. Type "test" in the terminal.
+
+This example consists of three tests.
+First will show you how you can box accounts
+Seconds shows how to create a 10kb account and fill it with data
+Third one shows how you can create accounts up to 10mb using zero copy serialization.
+
+## Explanation of Solana Memory and Zero Copy
+
+The heap and stack memory in the Solana runtime are very limited. We have 4Kb to work with on the stack and 32Kb on the heap.
+The stack increased by 10Kb per loaded account. These limits are quickly reached when writing a program.
+By default in Anchor all account structs are being loaded will be on the stack. If you reach the stack limit you will an error similar to this:
+
+```sh
+Stack offset of -30728 exceeded max offset of -4096 by 26632 bytes, please minimize large stack variables
+```
+
+(See test stacksize.ts)
+
+To prevent this to a certain degree you can Box your account. What this means is that the account will move to the heap and there will only a pointer of 16 Bytes will be saved on the Stack.
+This can be done like this:
+
+```rs
+#[derive(Accounts)]
+pub struct Example {
+    pub my_acc: Box<Account<'info, MyData>>
+}
+```
+
+If your account gets bigger it gets a bit more complicated. Solana does not allow Cross Program Invocations with accounts bigger than 10Kb.
+Anchor does use a CPI to initialize all new accounts. So it calls the System Program internally to create a new account.
+You can allocate more memory to your account like this with an extra transaction:
+
+Program:
+
+```rs
+
+    #[derive(Accounts)]
+    #[instruction(len: u16)]
+    pub struct IncreaseAccoutSize<'info> {
+        #[account(mut,
+            realloc = len as usize,
+            realloc::zero = true,
+            realloc::payer=signer)]
+        pub data_holder: Account<'info, DataHolderNoZeroCopy>,
+        #[account(mut)]
+        pub signer: Signer<'info>,
+        #[account(address = system_program::ID)]
```

Comment:
Specifying the type `Program<'info, System>` already does this check, so checking again is redundant.

```suggestion
```